http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -222,7 +223,8 @@ protected String getRedirectUri(String state) {
             String redirectUri = rewrittenRedirectUri(url);
             URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl());
             redirectUriBuilder.addParameter(RESPONSE_TYPE, CODE)
-                    .addParameter(CLIENT_ID, deployment.getResourceName());
+                    .addParameter(CLIENT_ID, deployment.getResourceName())
+                    .addParameter(NONCE, String.valueOf(state.hashCode()));
```

Comment:
We shouldn't be re-using the state value here. We should be using a separate value for the nonce.